burst/src/main/java/com/squareup/burst/TestConstructor.java

```diff
@@ -113,4 +114,17 @@ public void initializeFieldsOnInstance(final Object instance, final Object[] arg
       fields[i].set(instance, args[i]);
     }
   }
+
+  public Annotation[][] getArgumentAnnotations() {
+    final Annotation[][] ctorAnnotations = constructor.getParameterAnnotations();
```

Comment:
I'd prefer if this were inlined in the arraycopy call, and using `constructor.getParameterCount()` as the number of annotations